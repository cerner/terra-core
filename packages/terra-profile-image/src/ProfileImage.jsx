import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import TerraImage from 'terra-image';
import styles from './ProfileImage.module.scss';

const cx = classNames.bind(styles);

/* eslint react/no-unused-prop-types: [0] */
const propTypes = {
  /**
   * The source for the image which will be displayed.
   */
  src: PropTypes.string,
  /**
   * The text content that specifies an alternative text for an image.
   */
  alt: PropTypes.string,
  /**
   * Sets the height of the image.
   */
  height: PropTypes.string,
  /**
   * Sets the width of the image.
   */
  width: PropTypes.string,
  /**
   * Function to be executed when the profile image load is successful.
   */
  onLoad: PropTypes.func,
  /**
   * Function to be executed when the profile image load errors.
   */
  onError: PropTypes.func,
};

const isOnlyNumbers = toTest => !(/\D/).test(toTest);

const ProfileImage = (props) => {
  // img tags assume a height attribute of only numbers is in px but CSS does not
  const fixedHeight = isOnlyNumbers(props.height) ? `${props.height}px` : props.height;
  const fixedWidth = isOnlyNumbers(props.width) ? `${props.width}px` : props.width;

  const placeholderImage = (
    <span
      className={cx('placeholder-images')}
      title={props.alt}
      style={{ height: fixedHeight, width: fixedWidth }}
    />
  );

  if (props.src) {
    return (<div><TerraImage placeholder={placeholderImage} {...props} /></div>);
  }
  return placeholderImage;
};

ProfileImage.propTypes = propTypes;

export default ProfileImage;
