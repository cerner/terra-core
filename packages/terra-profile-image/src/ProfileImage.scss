/* stylelint-disable selector-class-pattern */
:local {
  .hidden {
    display: none;
  }
}
