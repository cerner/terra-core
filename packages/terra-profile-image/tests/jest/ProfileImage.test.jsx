import React from 'react';
import ProfileImage from '../../src/ProfileImage';

/* Default Profile Image */
it('should render a Terra image with a placeholder prop', () => {
  const wrapper = shallow(<ProfileImage src="profile.jpg" height="75" width="75" alt="A profile image" />);
  expect(wrapper).toMatchSnapshot();
});

/* Profile Image source is not provided */
it('should render an avatar image without Source prop', () => {
  const wrapper = shallow(<ProfileImage alt="Default" height="75" width="75" />);
  expect(wrapper).toMatchSnapshot();
});
