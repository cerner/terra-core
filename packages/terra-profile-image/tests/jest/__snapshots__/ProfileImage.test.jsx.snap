// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a Terra image with a placeholder prop 1`] = `
<div>
  <Image
    alt="A profile image"
    height="75"
    isFluid={false}
    placeholder={
      <span
        className="placeholder-images"
        style={
          Object {
            "height": "75px",
            "width": "75px",
          }
        }
        title="A profile image"
      />
    }
    src="profile.jpg"
    variant="default"
    width="75"
  />
</div>
`;

exports[`should render an avatar image without Source prop 1`] = `
<span
  className="placeholder-images"
  style={
    Object {
      "height": "75px",
      "width": "75px",
    }
  }
  title="Default"
/>
`;
